src/figuresClasses.ts

```diff
@@ -1,19 +1,86 @@
-export interface Figure {
+type Color = 'red' | 'green' | 'blue';
+
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+function hasZeroSize(...lengthes: number[]): void {
+  const hasNullSize = lengthes.some((number) => number <= 0);
 
+  if (hasNullSize) {
+    throw new Error('each side or radius must be greater than zero');
+  }
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    hasZeroSize(a, b, c);
+
+    const maxSide = Math.max(a, b, c);
+    const sidesSum = a + b + c;
+    const hasImpossibleSize = maxSide >= sidesSum - maxSide;
 
+    if (hasImpossibleSize) {
+      throw new Error('biggest side can\'t be greater that sum of another two');
+    }
+  }
+
+  getArea(): number {
+    const p = (this.a + this.b + this.c) / 2;
+    const pA = (p - this.a);
+    const pB = (p - this.b);
+    const pC = (p - this.c);
```

Comment:
Don't know what are `p`, `pA`... :harold:

Reply:
its perimeter in math)